starter-code/src/clue.js

```diff
@@ -1,94 +1,152 @@
+last_name:    "Peacock",
+color:        "blue",
+description:  "She is from a wealthy family and uses her status and money to earn popularity",
+age:          36,
+image:        "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+occupation:   "Socialité"
+}
+var mrMustard = {
+first_name:   "Jack",
+last_name:    "Mustard",
+color:        "yellow",
+description:  "He is a former football player who tries to get by on his former glory",
+age:          62,
+image:        "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+occupation:   "Retired Football player"
+}
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
 
+// Weapons
+var weapons = [
+{name: "rope" , weight: 10},
+{name: "knife"     ,   weight: 8},
+{name: "candlestick" , weight: 2},
+{name: "dumbbell"   ,   weight: 30},
+{name: "poison" ,   weight: 2},
+{name: "axe"  ,   weight: 15},
+{name: "bat"   ,   weight: 13},
+{name: "trophy"   ,   weight: 25},
+{name: "pistol" ,   weight: 20},
+]
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+var rooms = [
+{name: "Dinning Room"},
+{name: "Conservatory"},
+{name: "Kitchen"},
+{name: "Study"},
+{name: "Library"},
+{name: "Billiard Room"},
+{name: "Lounge"},
+{name: "Ballroom"},
+{name: "Hall"},
+{name: "Spa"},
+{name: "Living Room"},
+{name: "Observatory"},
+{name: "Theater"},
+{name: "Guest House"},
+{name: "Patio"}
+]
 
 // Characters Collection
 var charactersArray = [];
-
+charactersArray.push(mrGreen, mrMustard,mrsPeacock,missScarlet,drOrchid,profPlum)
 // Rooms' Collection
 var roomsArray = [];
-
+roomsArray.push(rooms)
 // Weapons Collection
 var weaponsArray = [];
+weaponsArray.push(weapons)
+
+
+function randomSelector(arr) {
+    var randomIndex = Math.floor(Math.random() * arr.length);
+    return arr[randomIndex];
+  }
+  
+  
+  
+  function pickMistery() {
+        var misteryEnvelope = [
+          randomSelector(charactersArray), 
+          randomSelector(weaponsArray), 
+          randomSelector(roomsArray)
+        ];
+      
+     
+      return misteryEnvelope;
+    }
+//   pickMistery();
+
+
+  function revealMistery(misteryEnvelope) {
+    // var simpleArray = [''];
+//     var simpleArray = `${c`haractersArray.first_name} ${charactersArray.last_name} killed Mr.Boddy using the ${weaponsArray} in the ${roomsArray}`;
+// return simpleArray;
+var simpleArray = `${misteryEnvelope[0].first_name} ${misteryEnvelope[0].last_name} killed Mr.Boddy using the ${misteryEnvelope[1].name} in the ${misteryEnvelope[2].name}!!!!`;
```

Comment:
Every variable that you create takes part of the memory so try not to create variables when there's no need. You can just:
```
return `${misteryEnvelope[0].first_name} ${misteryEnvelope[0].last_name} killed Mr.Boddy using the ${misteryEnvelope[1].name} in the ${misteryEnvelope[2].roomNames}!!!!`
```